app/controllers/tasks_controller.rb

```diff
@@ -2,17 +2,43 @@
 
 class TasksController < ApplicationController
   def index
-    # TODO: ad-hoc goal
     # TODO: goal completion
-    @date = date_from_params || current_user.beginning_of_day
-    @task_list = RetrieveTodaysTaskListAction.new(user: current_user).call(today: @date)
-    @task_list ||= GenerateTodaysTaskListAction.new(user: current_user).call(today: @date)
-    authorize @task_list
-    @task_list
+    # TODO: goal ordering
+
+    if params[:task_list_id].nil?
+      @date = date_from_params || current_user.beginning_of_day
+      task_list = task_list_from_date(date: @date)
+    else
+      task_list = task_list_from_id(id: params[:task_list_id])
+
+      if task_list.nil?
+        redirect_to tasks_path
+        return
+      end
+
+      @date = task_list.date
+    end
```

Comment:
can this be extracted into something like `[task_list, date] = get_task_list_and_date`

Reply:
Or a request/dto object

Reply:
Or return a hash